src/index/base.h

```diff
@@ -75,8 +76,9 @@ class BaseIndex : public CValidationInterface
     /// to a chain reorganization), the index must halt until Commit succeeds or else it could end up
     /// getting corrupted.
     bool Commit();
-
 protected:
+    CChainState* m_chainstate{nullptr};
```

Comment:
nit: Add comment "Provide access to the active chainstate, containing the current most-work chain". Maybe even as a member name `m_active_chainstate`

Reply:
Not sure `m_active_chainstate` makes too much sense here, as it's "the active chainstate at the time when `::Start` was called"

Reply:
Sure, it's just with assumeutxo, we start to have multiple chainstate references through the codebase (`m_ibd_chainstate`, `m_snapshot_chainstate`, `m_m_active_chainstate`) and it could be confusing in the future. I don't think it changes anything for the indexes rn.

Reply:
To @ariard's point, we will likely have to facilitate changing this pointer mid-execution when the user calls `ActivateSnapshot()`. I'll have to warm my caches on the indexing stuff a little more, but IIRC we want this class instance to have a reference to the current active chainstate (which obviously will be liable to change as assumeutxo progresses).

Reply:
That said, the reduction of global use is an obvious win, and it should be pretty easy to facilitate changing `m_chainstate` via some method that can be called by `ActivateSnapshot()`.

Reply:
Might even just make sense to pass in a `ChainstateManager` instance to keep with this class... but that can be done in a follow-up.